### PR TITLE
bpo-35416: Fix potential resource warnings in distutils

### PR DESCRIPTION
**Pull request by BoboTiG**
Prevent file descriptor leaks in distutils.

https://bugs.python.org/issue35416

### CODE REVIEW

**Comment by merwok** [comment]
Thanks, this looks good!

There are a few instances of weird indentation (from before your patch), I would like to take the opportunity to clean them.

**Comment by BoboTiG** [comment]
Of course, go ahead :)
Or do you want me to do the clean?

**Comment by merwok** [comment]
Easier if I do them!  It could take me a couple weeks though.

Another PR with some of the fixes was just merged, so part of the codebase is already better.

**Comment by BoboTiG** [comment]
Good, we can close the issue then :)

**Review by serhiy-storchaka — APPROVED** [review]

File: Lib/distutils/command/bdist_wininst.py [hunk 1]
```diff
@@ -247,47 +247,49 @@ def create_exe(self, arcname, fullname, bitmap=None):
...
-                             )
-        file.write(header)
-        file.write(open(arcname, "rb").read())
+            if self.pre_install_script:
+                # We need to normalize newlines, so we open in text mode and
+                # convert back to bytes. "latin-1" simply avoids any possible
+                # failures.
+                with open(self.pre_install_script, "r",
+                    encoding="latin-1") as script:
+                    script_data = script.read().encode("latin-1")
```
Comment (merwok) [inline Lib/distutils/command/bdist_wininst.py:273]: @methane I wanted to fix this bad/weird indentation and didn’t have the time

Comment (methane) [reply Lib/distutils/command/bdist_wininst.py:273]: fixed in master.

**Pull request merged**